lib/src/Uncertainty/Algorithm/Optimization/EfficientGlobalOptimization.cxx

```diff
@@ -302,14 +302,39 @@ void EfficientGlobalOptimization::run()
     }
     solver_.run();
     const OptimizationResult improvementResult(solver_.getResult());
-
     // store improvement
-    Point improvementValue(improvementResult.getOptimalValue());
-    expectedImprovement_.add(improvementValue);
+    Point newPoint(improvementResult.getOptimalPoint());
+    Point newValue;
+    Point improvementValue;
+    if (!problem.hasBounds() || problem.getBounds().contains(newPoint))
+      {
+        improvementValue = improvementResult.getOptimalValue();
+        expectedImprovement_.add(improvementValue);
+
+        newValue = model(newPoint);
+        ++ evaluationNumber;
+      }
+    else
+    // Restore the admissibility wrt bounds if needed
```

Comment:
This should not be needed if enforced in other algos, see if my last commit in misc2 does what you want